libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,751 @@
+import static uk.gov.di.ipv.core.library.service.CiMitService.GOVUK_SIGNIN_JOURNEY_ID_HEADER;
+import static uk.gov.di.ipv.core.library.service.CiMitService.IP_ADDRESS_HEADER;
+import static uk.gov.di.ipv.core.library.service.CiMitService.POST_CI_ENDPOINT;
+import static uk.gov.di.ipv.core.library.service.CiMitService.POST_MITIGATIONS_ENDPOINT;
+
+@ExtendWith(PactConsumerTestExt.class)
+@ExtendWith(MockitoExtension.class)
+@PactTestFor(providerName = "CiMitProvider")
+@MockServerConfig(hostInterface = "localhost")
+class ContractTest {
+    @Mock ConfigService mockConfigService;
+
+    @BeforeEach
+    void setUp() {
+        when(mockConfigService.enabled(CIMIT_API_GATEWAY_ENABLED)).thenReturn(true);
+    }
+
+    @Pact(provider = "CiMitProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact getCisUserIdReturnsContraIndicators(PactDslWithProvider builder) {
+        var responseForGetCi = newJsonBody(body -> body.stringValue("vc", VALID_CI_VC_JWT)).build();
+
+        return builder.given("mockUserId is the user_id")
+                .given("the current time is 2024-01-01 00:00:00")
+                .given("mockCimitComponentId is the issuer")
+                .given("a contra-indicator is returned with code TEST-CI-CODE-2 for a passport")
+                .given("the passport has issue date 2024-08-05T14:59:03.000Z")
+                .given("the passport has document number 12345678")
+                .given("the mitigation has code TEST02")
+                .given("the mitigation is valid from 2024-08-05T14:59:05.000Z")
+                .given("the mitigation has no incomplete mitigations")
+                .given("a contra-indicator is returned with code TEST-CI-CODE-1 for an id card")
+                .given("the id card has issue date 2024-08-05T14:59:04.000Z")
+                .given("the id card has document number 852654")
+                .given("the mitigation has code TEST01")
+                .given("the mitigation is valid from 2024-08-05T14:59:05.000Z")
+                .given("the mitigation has no incomplete mitigations")
+                .uponReceiving(
+                        "Request for contra-indicators for specific user with existing contra-indicators.")
+                .path(GET_VCS_ENDPOINT)
+                .query("user_id=" + MOCK_USER_ID)
+                .method("GET")
+                .headers(
+                        IP_ADDRESS_HEADER,
+                        MOCK_IP_ADDRESS,
+                        GOVUK_SIGNIN_JOURNEY_ID_HEADER,
+                        MOCK_GOVUK_SIGNIN_ID,
+                        PactDslRequestBase.CONTENT_TYPE,
+                        "application/json")
+                .willRespondWith()
+                .status(200)
+                .body(responseForGetCi)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "getCisUserIdReturnsContraIndicators")
+    void fetchContraIndicators_whenCalledWithUserIdAgainstCimiApi_receivesContraIndicators(
```

Comment:
```suggestion
    void fetchContraIndicators_whenCalledWithUserIdAgainstCimitApi_receivesContraIndicators(
```

& below Cimit's are missing their "t"'s

Reply:
good spot!